tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -949,7 +949,7 @@ void EnqueueProgramCommand::assemble_device_commands(
     const uint32_t max_length_per_sub_cmd = dispatch_constants::get(this->dispatch_core_type).scratch_db_size() / 2;
     const uint32_t max_paged_length_per_sub_cmd =
         max_length_per_sub_cmd / HostMemDeviceCommand::PROGRAM_PAGE_SIZE * HostMemDeviceCommand::PROGRAM_PAGE_SIZE;
-    const auto &kernels_buffer = program.get_kernels_buffer();
+    const auto kernels_buffer = program.get_kernels_buffer(this->device);
```

Comment:
`get_kernels_buffer(this->device)` based on your current impl can return nullptr and you're dereferencing without a check... should you throw an exception if your expectation is that it is always non-null

Reply:
Added an assert